kernel/src/table_properties/deserialize.rs

```diff
@@ -36,20 +34,20 @@ where
 
 // attempt to parse a key-value pair into a `TableProperties` struct. Returns Some(()) if the key
 // was successfully parsed, and None otherwise.
-fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
+fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Result<(), &'static str> {
```

Comment:
Option here always generally seemed like the wrong type here. It would make sense to replace this with `Result<(), SomeNewError>` instead.